pkg/engine/jsonContext.go

```diff
@@ -183,6 +184,9 @@ func fetchConfigMap(entry kyverno.ContextEntry, lister dynamiclister.Lister) ([]
 		return nil, fmt.Errorf("failed to convert configmap %s/%s: %v", namespace, name, err)
 	}
 
+	// update the unstructuredObj["data"] to delimit and split the string value (containing "\n") with "\n"
+	unstructuredObj["data"] = mapRefactorizer(unstructuredObj["data"].(map[string]interface{}))
```

Comment:
should this logic be co-located with where we parse the JSON String format?

```yaml
data:
  allowed-roles: "[\"cluster-admin\", \"cluster-operator\", \"tenant-admin\"]"
```

We can move that logic here if needed, to help maintain (or move this to where that is handled.)

Reply:
The JSON formatted value `["cluster-admin", "cluster-operator", "tenant-admin"]` will be parsed as a single string in the context, it will be then parsed to the slice during the evaluation, for example:
https://github.com/kyverno/kyverno/blob/da8e449d3cb586e139655ca616cbfefb565442e4/pkg/engine/variables/operator/in.go#L86-L103

Reply:
Yes, Shuting is right. The helper function which I wrote will basically delimit/split the body by "\n". So, `"[\"cluster-admin\", \"cluster-operator\", \"tenant-admin\"]"` will be interpreted as `["[\"cluster-admin\", \"cluster-operator\", \"tenant-admin\"]"]` and further execution will be exactly the same.